### ISSUE TITLE
Better choice of loss function for hyperparameter block

### ISSUE THREAD

**Issue by ToonTalk** [ecraft2learn/ai#35]
SInce categoricalCrossEntropy is better for classification problems maybe look at the size of final layer and if greater than 1 prefer this one otherwise don't consider it.
